design-proposals/primary_network_binding_for_seamless_migration.md

```diff
@@ -0,0 +1,459 @@
+| Deployment                  | Binding Plugin                                                                                                  | Core                                 | Binding Plugin                                   |
+| IP & Route delivery         | Self (IPv4 & IPV6)                                                                                              | Self for IPv4, Net Provider for IPv6 | Net Provider                                     |
+| Seamless Live-Migration     | Requires Development                                                                                            | Yes                                  | Yes                                              |
+| Supported Traffic           | TCP/UDP/ICMP (All IP in case seamless migration is resolved)                                                    | IP                                   | IP                                               |
+| Migration Support Indicator | Yes                                                                                                             | Requires Development                 | Yes                                              |
+| Mesh / Istio support        | Yes                                                                                                             | No                                   | No                                               |
+| IP Status (reporting)       | Yes                                                                                                             | Yes                                  | Requires Development                             |
+| Resource Overhead           | 500MB per VM, 250MB per VM already available on U/S, Negligible overhead when resolving the seamless migration. | Negligible                           | Negligible                                       |
+| Performance                 | N/A                                                                                                             | N/A                                  | N/A , Most likely better than the bridge option. |
+| Kubevirt Support            | Low                                                                                                             | High                                 | Medium                                           |
+
+[*] Core Bridge binding can be deployed as a binding plugin with a subset
+    of its existing feature sets. E.g. as a plugin, IP & routes can be
+    supported only through the network-provider.
+
+### The Binding Winner
+
+Considering the requirements, development effort and maintenance cost,
+the following plan is chosen:
+1. Develop a **macvtap** network binding plugin.
+2. In case option [1] fails, develop a **bridge** network binding plugin.
+3. In case option [2] fails, enhance the existing **bridge** core binding.
+4. Develop and enhance the Passt network binding plugin for the mid/long term.
```

Comment:
I think this do not have to wait for 1, 2 and 3, ideally we move to passt in the future so it can be developed in parallel.

Reply:
huge +1.